server/src/main/java/org/elasticsearch/threadpool/ThreadPool.java

```diff
@@ -59,13 +59,22 @@
 import static java.util.Map.entry;
 import static org.elasticsearch.core.Strings.format;
 
+/**
+ * Manages all the Java thread pools we create. {@link Names} contains a list of the thread pools, but plugins can dynamically add more
+ * thread pools to instantiate.
+ */
 public class ThreadPool implements ReportingService<ThreadPoolInfo>, Scheduler {
 
     private static final Logger logger = LogManager.getLogger(ThreadPool.class);
 
     public static class Names {
+        // Often used for quick operations not worth the costs of switching to another thread. Also used to avoid the cost of re-queuing
+        // work while leveraging the interface for ease of coding.
+        // This should not be used on Netty transport threads if the task may not always complete quickly.
```

Comment:
Could we make these comments JavaDoc? That way they'll be more easily visible in IDEs.